27/resources/page

ccoutts2 opened the pull request:
- Add resources component to resources page
- Toggle state to change which articles are rendered
- Add gsap animation to articles header

ccoutts2 committed:
WIP - create dynamic article filtering using useState

ccoutts2 committed:
Merge branch 'develop' into 27/resources/page

ccoutts2 committed:
Add gsap animation to article headers, add lenis scroll to page

kabirseth4 commented on src/app/resources/page.tsx:
```diff
@@ -1,26 +1,65 @@
-      <div>page</div>
+      <PageHeader header="resources" />
+      <section className="centered flex-col">
+        <nav>
```
Do we need this nav while there are only 3 articles? Don't think Holly is planning to write anymore for the time being so maybe we can just leave this commented out. Also don't think we need the section wrapper, I think just nav is fine.

ccoutts2 replied:
Yeah I'm not sure why I used a nav link in the first place. I think in my head I thought the contents act as a navigation menu to another page. Article tag would be better.

I think I'm going to come back to this branch another time and render the contents of the articles server side. We can have a chat about this at some point but I think creating the articles as markdown files is the best way forward.

kabirseth4 commented on src/components/Resources/data.ts:
```diff
@@ -0,0 +1,52 @@
+interface articlesProps {
```
Capitalise, and not props. Maybe just Article

kabirseth4 commented on src/components/Resources/Resources.tsx:
```diff
@@ -1,42 +1,35 @@
+}
+
+const Resources = ({ showHeader = true, articleFilter }: ResourcesProps) => {
+  const filteredArticles = articles.filter(
```
Again, not sure if needed for now. We can always set a default value for articleFilter for time being and leave rest as is

kabirseth4 reviewed (changes requested):
Unsure about need of article filter functionality

ccoutts2 commented:
@kabirseth4 

As commented above, I think its best to render the contents of the article server side, of which the content will be in markown file. I'll restructure the files accordingly. 

As for your comment on the filter functionality, this is because the same component is used in the home page and I wanted to display only 3 articles from what will be a longer list in future. I created dummy data in a file in the resources component as if there were more articles. It would make more sense to render these by latest articles rather than title, so will set up the functionality for this

ccoutts2 committed:
Merge branch 'develop' into 27/resources/page

ccoutts2 committed:
Setup logic for converting markdown to html, create dynamin routing

ccoutts2 committed:
Add image to markdown meta data

ccoutts2 committed:
Refactor resources component to render the content from markdown

ccoutts2 commented:
Changes:

- Resources (blogs) are now converted from markdown and rendered server side
- Dynamic routing for each blog set up
- Removed category filtering and can think about how we set it up when we review the pages
- Removed lenis scroll from Home page to enable server-side rendering of the page

@kabirseth4 - styling is not great but rather than waste time playing around with it, let's get the rest of the website built before we think about new sytling / page layouts

ccoutts2 committed:
Remove data file in resources component

ccoutts2 committed:
Create ArticleCard component

ccoutts2 committed:
Add server side url searchParams filtering

ccoutts2 committed:
Set limit on how many posts to show on home page

ccoutts2 commented:
@kabirseth4 
Changes:

- Moved article cards to a card component
- Adjusted some styling
- Implemented useSearchParams to filter through different categories of articles
- Limit of 3 posts for home page

ccoutts2 committed:
Merge branch 'develop' into 27/resources/page
